src/components/SignupForm/SignupForm.jsx

```diff
@@ -1,22 +1,101 @@
 import BadgeIcon from '@mui/icons-material/Badge'
-import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline'
 import LockRoundedIcon from '@mui/icons-material/LockRounded'
 import PersonIcon from '@mui/icons-material/Person'
-import PhoneIphoneIcon from '@mui/icons-material/PhoneIphone'
-import RadioButtonUncheckedIcon from '@mui/icons-material/RadioButtonUnchecked'
-import { Checkbox, InputAdornment, Link, Tab, Tabs } from '@mui/material'
+import VisibilityIcon from '@mui/icons-material/Visibility'
+import VisibilityOffIcon from '@mui/icons-material/VisibilityOff'
+import {
+    Checkbox,
+    IconButton,
+    InputAdornment,
+    Link,
+    Tab,
+    Tabs,
+} from '@mui/material'
+import { useState } from 'react'
 import * as styleMui from './SignupForm.styled'
 
 const iconStyle = {
     color: '#69AD28',
     fontSize: '2rem',
 }
 
+const passValidation =
+    /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9])(?!.*\s).{6,15}$/
+
+const emailValidation = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/
+
 export default function SignupForm({ setTypeUser, typeUser }) {
     const handleChange = (event, newValue) => {
         setTypeUser(newValue)
     }
 
+    //Bắt lỗi input
+    const [email, setEmail] = useState('')
+    const [firstName, setFirstName] = useState('')
+    const [lastName, setLastName] = useState('')
+    const [password, setPassword] = useState('')
+    const [reEnterPassword, setReEnterPassword] = useState('')
+    const [emailErrorText, setEmailErrorText] = useState('')
+    const [firstNameErrorText, setFirstNameErrorText] = useState('')
+    const [lastNameErrorText, setLastNameErrorText] = useState('')
+    const [passwordErrorText, setPasswordErrorText] = useState('')
+    const [reEnterPasswordErrorText, setReEnterPasswordErrorText] = useState('')
+    const [isTrue, setIsTrue] = useState(false)
+    const [isTrueErrorText, setIsTrueErrorText] = useState('')
```

Comment:
So, its still good, but if we have to many state here then we can change state to object, for example : const [state, setState] = useState({vldPassword:value, vldUserName:vl, vldSubName:vl,.............You can add more here })